src/legacy/core_plugins/kibana/public/visualize/editor/editor.js

```diff
@@ -373,8 +393,19 @@ function VisEditor(
       }
     };
 
-    timefilter.enableAutoRefreshSelector();
+    const updateRefreshInterval = () => {
+      $scope.refreshInterval = timefilter.getRefreshInterval();
+      // In case we are running in embedded mode (i.e. we used the visualize loader to embed)
+      // the visualization, we need to update the refreshInterval on the visualize handler.
+      if ($scope._handler) {
+        $scope._handler.update({
+          refreshInterval: $scope.refreshInterval,
```

Comment:
@timroes do I need to do this here? I'm copying what the `updateTimeRange` function above does but I'm not really familiar with what this `_handler` is doing.

Reply:
@Bargs That is not needed there. The [`update`](https://github.com/elastic/kibana/blob/master/src/ui/public/visualize/loader/embedded_visualize_handler.ts#L199) only accepts `timeRange`, `dataAttrs`, `filters` and `query` as params (see https://github.com/elastic/kibana/blob/master/src/ui/public/visualize/loader/types.ts#L123:L126).

We don't need to tell the visualization the `refreshInterval` at all, since the refresh in case that interval is triggered, is not done by the visualization itself, but it will be triggered from the outside (at the moment via a very weird way through the courier search poll, emitting events, that `vis` will listen on).

Reply:
Thanks for the quick response Tim, I've removed those lines 👍